provider/pkg/await/await.go

```diff
@@ -491,9 +499,69 @@ func ssaUpdate(c *UpdateConfig, liveOldObj *unstructured.Unstructured, client dy
 	return currentOutputs, nil
 }
 
+// handleSSAIgnoreFields handles updating the inputs to either drop fields that are present on the cluster and not managed
+// by the current field manager, or to set the value of the field to the last known value applied to the cluster.
+func handleSSAIgnoreFields(c *UpdateConfig, liveOldObj *unstructured.Unstructured) error {
+	managedFields := liveOldObj.GetManagedFields()
+	managedFieldsSet := fieldpath.NewSet()
+
+	for _, f := range managedFields {
+		if f.Manager == c.FieldManager {
+			// We don't need to keep track of fields that we are currently managing.
+			continue
+		}
+		s := fieldpath.NewSet()
+		if err := s.FromJSON(bytes.NewReader(f.FieldsV1.Raw)); err != nil {
+			return fmt.Errorf("unable to parse managed fields from resource %q: %w", c.Inputs.GetName(), err)
+		}
+		managedFieldsSet = managedFieldsSet.Union(s)
+	}
+
+	for _, ignorePath := range c.IgnoreChanges {
+		pathComponents := strings.Split(ignorePath, ".")
```

Comment:
It looks like this is the ignore changes Pulumi Resource Option.

If so, valid paths are more complicated than this, for example `foo.bar[3].fizz[*]`. We should use the [library provided in `pulumi/sdk`](https://github.com/pulumi/pulumi/blob/4ebf61f896fe226bf455da3fc87c91abdc33dc28/sdk/go/common/resource/properties_path.go#L49) for parsing.

Reply:
To @iwahbe's point, I observe that `Diff` uses `resource.ParsePropertyPath(ignore)` to parse the `IgnoreChanges`.